packages/fast-usdc/src/fast-usdc.contract.js

```diff
@@ -168,19 +168,22 @@ export const contract = async (zcf, privateArgs, zone, tools) => {
       return poolKit.feeRecipient.makeWithdrawFeesInvitation();
     },
     /**
-     * @param {string} agoricChainId
-     * @param {string} nobleChainId
-     * @param {IBCConnectionInfo} agoricToNoble
+     * @param {string} [agoricChainId]
+     * @param {string} [nobleChainId]
+     * @param {IBCConnectionInfo} [agoricToNoble]
      */
     async connectToNoble(agoricChainId, nobleChainId, agoricToNoble) {
-      trace('connectToNoble', agoricChainId, nobleChainId, agoricToNoble);
-      chainHub.updateConnection(agoricChainId, nobleChainId, agoricToNoble);
+      const shouldUpdate = agoricChainId && nobleChainId && agoricToNoble;
+      if (shouldUpdate) {
+        trace('connectToNoble', agoricChainId, nobleChainId, agoricToNoble);
+        chainHub.updateConnection(agoricChainId, nobleChainId, agoricToNoble);
+      }
+      const nobleICALabel = `NobleICA-${(shouldUpdate ? agoricToNoble : agToNoble).counterparty.connection_id}`;
```

Comment:
Can we be sure at this point that `agToNoble` has settled?  It's awaited below, but that's after the synchronous prelude, and the contract is reachable before there's any guarantee that the async stuff is done.

Reply:
Good question. We expect this to resolve promptly because: 1) `chainHub` is in the same vat, and 2) `chainHub.getChainsAndConnection` will not result in a call to the `agoricNames` vat as the information is cached in state.

In incarnation 2, our successful `makeSettler` call also evidences that `agToNoble` is settled before `start` is done. The `update-settler-reference` proposal waits for `upgradeContract` / `start` to complete before calling `connectToNoble`.